drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -746,7 +763,13 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		/* Start transaction */
 		i2c_transaction(dev);
 		/* Wait for the transfer to complete */
-		k_sem_take(&data->device_sync_sem, K_FOREVER);
+		/* TODO: the timeout should be adjustable */
+		res = k_sem_take(&data->device_sync_sem, K_MSEC(100));
+		if (res != 0) {
+			data->err = ETIMEDOUT;
+			/* reset i2c port */
+			i2c_reset(dev, I2C_RC_TIMEOUT);
```

Comment:
Why continue with the next message if this one failed?

Reply:
Yes, this transaction should be dropped. It's fixed.